content/influxdb/cloud-serverless/reference/client-libraries/v3/java.md

```diff
@@ -15,9 +14,306 @@ aliases:
+      System.err.println("Failed to query data from the bucket.");
+      e.printStackTrace();
+    }
+  }
+}
+```
+
+{{% cite %}}Source: [suyashcjoshi/SimpleJavaInfluxDB](https://github.com/suyashcjoshi/SimpleJavaInfluxDB/) on GitHub{{% /cite %}}
+
+Replace the following:
+
+- {{% code-placeholder-key %}}`DATABASE_NAME`{{% /code-placeholder-key %}}: your {{% product-name %}} [bucket](/influxdb/cloud-serverless/admin/buckets/)
+- {{% code-placeholder-key %}}`API_TOKEN`{{% /code-placeholder-key %}}: a local environment variable that stores your [token](/influxdb/cloud-serverless/admin/tokens/)--the token must have read and write permission to the specified bucket.
+
+### Run the example to write and query data
+
+1. Build an executable JAR for the project--for example, using Maven:
+   
+   <!--pytest.mark.skip-->
+
+   ```bash
+   mvn package
+   ```
+   
+2. In your terminal, run the `java` command to write and query data in your bucket:
+
+   <!--pytest.mark.skip-->
+
+   ```bash
+   java \
+   --add-opens=java.base/java.nio=org.apache.arrow.memory.core,ALL-UNNAMED \
+   -jar target/PROJECT_NAME.jar
+   ```
+
+   Include the following in your command:
+
+   - [`--add-opens=java.base/java.nio=org.apache.arrow.memory.core,ALL-UNNAMED`](https://arrow.apache.org/docs/java/install.html#id3): with Java version 9 or later and Apache Arrow version 16 or later, exposes JDK internals for Arrow.
+   For more options, see the [Apache Arrow Java install documentation](https://arrow.apache.org/docs/java/install.html).
+   - `-jar target/PROJECT_NAME.jar`: your `.jar` file to run.
+
+The output is the newly written data from your {{< product-name >}} bucket.
+
+## Installation
+
+Include `com.influxdb.influxdb3-java` in your project dependencies.
+
+{{< code-tabs-wrapper >}}
+{{% code-tabs %}}
+[Maven pom.xml](#)
+[Gradle dependency script](#)
+{{% /code-tabs %}}
+{{% code-tab-content %}}
+```xml
+<dependency>
+  <groupId>com.influxdb</groupId>
+  <artifactId>influxdb3-java</artifactId>
+  <version>RELEASE</version>
+</dependency>
+```
+{{% /code-tab-content %}}
+{{% code-tab-content %}}
+<!--pytest.mark.skip-->
+```groovy
+dependencies {
+
+   implementation group: 'com.influxdb', name: 'influxdb3-java', version: 'latest.release'
+
+}
+```
+{{% /code-tab-content %}}
+{{< /code-tabs-wrapper >}}
+
+## Importing the client
+
+The `influxdb3-java` client library package provides
+`com.influxdb.v3.client` classes for constructing, writing, and querying data
+stored in {{< product-name >}}.
+
+## API reference
+
+- [Interface InfluxDBClient](#interface-influxdbclient)
+  - [Initialize with credential parameters](#initialize-with-credential-parameters)
+  - [InfluxDBClient instance methods](#influxdbclient-instance-methods)
+  - [InfluxDBClient.writePoint](#influxdbclientwritepoint)
+  - [InfluxDBClient.query](#influxdbclientquery)
+
+
+## Interface InfluxDBClient
+
+Provides an interface for interacting with InfluxDB APIs for writing and querying data.
```

Comment:
With the headings, I'm trying to follow common patterns in other client API docs that I like, but it really needs some CSS work to differentiate the _type_ from the name.

Reply:
We could definitely add that in. We could add API-specific heading classes. Do you have examples of what you would want? Does the type need to be included in the DOM or could it be a psuedo element?

Reply:
> We could definitely add that in. We could add API-specific heading classes. Do you have examples of what you would want? Does the type need to be included in the DOM or could it be a psuedo element?

The main reason I've stalled on it is that I wanted to explore using tools to generate (to some extent) docs for those--it might be possible with Copilot now. If that did work, then I would expect it to generate Markdown and pseudo elements wouldn't be necessary.

OTOH, if we just want something that looks good now, then I think pseudo would be fine. Would SEO be hindered by that?

Originally, I was thinking something like [Arrow's reference docs](https://arrow.apache.org/docs/python/generated/pyarrow.flight.FlightClient.html), but documenting at that level of detail (with method overloads, etc) isn't manageable for us unless we generate it (like for Flux)--how far do we take it before just linking to the source in GitHub?

At a minimum, I think we need code format (``) in headings--for example: https://doc.rust-lang.org/rust-by-example/mod/use.html#the-use-declaration.

Beyond that, we could either:
- rely on code block highlighting for showing syntax or
- support emphasis (italics) and distinct colors for types (class, method, etc.), names, and argument values (like the [PyArrow example](https://arrow.apache.org/docs/python/generated/pyarrow.flight.FlightClient.html))